chia/wallet/wallet_node.py

```diff
@@ -466,38 +466,37 @@ async def _messages_to_resend(self) -> List[Tuple[Message, Set[bytes32]]]:
     async def _retry_failed_states(self):
         while not self._shut_down:
             try:
-                await asyncio.sleep(5)
-                current_time = time.time()
-                if self.last_state_retry_time < current_time - 10:
-                    self.last_state_retry_time = current_time
-                    if self.wallet_state_manager is None:
-                        continue
-                    states_to_retry = await self.wallet_state_manager.retry_store.get_all_states_to_retry()
-                    for state, peer_id, fork_height in states_to_retry:
-                        matching_peer = tuple(
-                            p for p in self.server.get_connections(NodeType.FULL_NODE) if p.peer_node_id == peer_id
-                        )
-                        if len(matching_peer) == 0:
-                            peer = self.get_full_node_peer()
-                            if peer is None:
-                                self.log.info(f"disconnected from all peers, cannot retry state: {state}")
-                                continue
-                            else:
-                                self.log.info(
-                                    f"disconnected from peer {peer_id}, state will retry with {peer.peer_node_id}"
-                                )
+                if self.wallet_state_manager is None:
+                    await asyncio.sleep(self.coin_state_retry_seconds)
```

Comment:
so, every `continue` in this loop is expected to `sleep()` first, is that right?

Reply:
Yes otherwise we might end up in a not sleeping, blocking endless loop.

Reply:
Okay well, now that we have the configurable timeout which is set to `1` for the test it doesn't really matter i guess, i added [00ea944](https://github.com/Chia-Network/chia-blockchain/pull/14911/commits/00ea944306865e94b3094eaa4ec65e686d2c7210).